### ISSUE TITLE
Users unable to export keys

### ISSUE THREAD

**Issue by xanrevent** [vector-im/riot-web#9459]
### Description

When a user tries to export keys via desktop or mobile, an error occurs.

### Steps to reproduce

This is a server hosted on Digital Ocean with Ubuntu 18.10 following the instructions at the following link: https://blog.cryptoaustralia.org.au/run-your-end-to-end-encrypted-chat-server-matrix-riot/ and also allowing http and https traffic through the firewall.

- On desktop
- When user attempts to sign out, they are prompted with the option to back up their keys
- User selects key backup, inputs and confirms password, and clicks download
- File is downloaded, but user is met with message "Starting backup.... unable to create key backup." 

- On mobile
- User  attempts logout and receives prompt
- User selects key backup option, inputs and confirms passphrase
- User receives error "Unexpected error / Unrecognized request"

On Mobile
- User attempts logout and receives prompt
- User selects manually export keys
- User inputs and confirms passphrase
- User receives error "Permission denied"

- that reproduce the bug
- using hyphens as bullet points

Expected behavior is successful export of security keys for a user.

Log: will be sent after the issue is submitted.

### Version information

- **Platform**: web (in browser) and mobile

For the web app:

- **Browser**: Chrome Version 73.0.3683.103 (Official Build) (64-bit)
- **OS**: Windows, TailsOS
- **URL**: private server , unknown version or riot-web

Mobile: android pie

**Comment by jryans** [comment]
@xanrevent What versions of Synapse and Riot are you using here?

**Comment by xanrevent** [comment]
I really want to answer this question, but I haven't been able to find the command to get the version numbers for either in the documentation for the last half hour. Would you kindly let me know which command to run? (this is slightly confounded by the fact that I thought I might be on an old version and so I purged the entire thing and tried a reinstall using sudo apt-get install matrix-synapse but have been unable to get the server working again due to problems trying to configure acme and nginx, so I'll report back if this was my fault for the version I had installed).

**Comment by jryans** [comment]
For Riot, there should be a `version` file in directory (if you downloaded a release version).

For Synapse, I would check via your package manager, so maybe `apt show matrix-synapse` should reveal something.

**Comment by mygitcrazy** [comment]
Hi there,

I noticed the same problem on my installation hosted on DigitalOcean droplet...
Synapse installed with command : "sudo apt install matrix-synapse -y"

Here are my version numbers:
Version: 0.24.0+dfsg-1
Priority: optional
Section: universe/net
Origin: Ubuntu 18.04


And the Riot app on windows:
riot-web-1.0.8

I just started with Matrix-Synapse and i like to help out with this as much as i can, because i am really enthusiast with the possibilities of it.

**Comment by jryans** [comment]
> Version: 0.24.0+dfsg-1

That's really quite old for Synapse, see the [releases](https://github.com/matrix-org/synapse/releases). See the Synapse README for installation [details](https://github.com/matrix-org/synapse/blob/master/INSTALL.md#prebuilt-packages). In particular, using the version available from Ubuntu directly is not recommended (as it's quite old).

In the absence of other info, it seems likely the first reporter got a very old Synapse the same way.

**Comment by jryans** [comment]
I would also suggest joining the #synapse:matrix.org room for help with Synapse deployment.

**Comment by mygitcrazy** [comment]
> I would also suggest joining the #synapse:matrix.org room for help with Synapse deployment.

Thanks for your quick reply's ;-)

Is it really that hard to install Matrix-Synapse the right way then?
I would like to ask somebody in the "synapse:matrix.org room" but isn't there just a good step-by-step tutorial somewhere to point to?

**Comment by jryans** [comment]
> Is it really that hard to install Matrix-Synapse the right way then?
> I would like to ask somebody in the "synapse:matrix.org room" but isn't there just a good step-by-step tutorial somewhere to point to? Something like this? --> https://upcloud.com/community/tutorials/install-matrix-synapse/
> Only thing in that tutorial would be to use python 3 instead of 2.7

The difficulty level depends on what you want, your environment, etc. I'd suggest reading through docs in the Synapse repo, chatting in that room, filing issues for new docs if you think they are needed, etc. See also https://github.com/matrix-org/matrix.org/issues/206.

**Comment by mygitcrazy** [comment]
> > Is it really that hard to install Matrix-Synapse the right way then?
> > I would like to ask somebody in the "synapse:matrix.org room" but isn't there just a good step-by-step tutorial somewhere to point to? Something like this? --> https://upcloud.com/community/tutorials/install-matrix-synapse/
> > Only thing in that tutorial would be to use python 3 instead of 2.7
> 
> The difficulty level depends on what you want, your environment, etc. I'd suggest reading through docs in the Synapse repo, chatting in that room, filing issues for new docs if you think they are needed, etc. See also [matrix-org/matrix.org#206](https://github.com/matrix-org/matrix.org/issues/206).

Tanks for the good pointers in that direction....
Too bad there isn't just a good and clear up-to-date-guide on this, because it is a nice piece of software to use.

**Comment by xanrevent** [comment]
I determined that my issue was also due to the apt installation I performed. I'm currently in the process of attempting an install via Docker and Ansible, but I'm also working on writing more thorough and in-depth instructions on the current installation manually for an Ubuntu installation on 18.04 and 19.04 with a few collaborators. 

Thank you everyone for the quick replies and thorough information! 

I was having other issues with the install, but I think that was due to the security issues and everything being moved around while I was trying to access it.
